vmdb/app/models/ems_openshift.rb

```diff
@@ -0,0 +1,34 @@
+class EmsOpenshift < EmsContainer
+  include ContainerProviderMixin
+
+  # TODO: what about k8s entities?
+  has_many :container_routes,                      :foreign_key => :ems_id, :dependent => :destroy
+  has_many :container_projects,                    :foreign_key => :ems_id, :dependent => :destroy
+
+  default_value_for :port, 8443
+
+  def self.ems_type
+    @ems_type ||= "openshift".freeze
+  end
+
+  def self.description
+    @description ||= "Openshift".freeze
+  end
+
+  def self.raw_connect(hostname, port)
+    require 'openshift_client'
+    api_endpoint = raw_api_endpoint(hostname, port)
+    osclient = OpenshiftClient::Client.new(api_endpoint)
+    # TODO: support real authentication using certificates
+    osclient.ssl_options(:verify_ssl => OpenSSL::SSL::VERIFY_NONE)
+    osclient
+  end
+
+  def kubernetes_connect
+    require 'kubeclient'
+    kube = Kubeclient::Client.new(api_endpoint)
+    # TODO: support real authentication using certificates
+    kube.ssl_options(:verify_ssl => OpenSSL::SSL::VERIFY_NONE)
+    kube
+  end
```

Comment:
Should this be a class method to match the raw_connect method. Something like `def self.raw_kubernetes_connect` or something.  Then, typically an instance that's service specific is exposed.  See something similar for the EmsAmazon.